### PR TITLE
Paywalls: Add support for `total_price_and_per_month` variable in paywalls

### PR DESCRIPTION
**Pull request by tonidero**
### Description
This adds support for the `total_price_and_per_month` variable in paywalls

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/TestData.kt [hunk 1]
```diff
@@ -296,7 +296,7 @@ private data class TestStoreProduct(
     private val introPrice: Price? = null,
 ) : StoreProduct {
     override val type: ProductType
-        get() = ProductType.SUBS
+        get() = if (period == null) ProductType.INAPP else ProductType.SUBS
```
Comment (NachoSoto) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/TestData.kt:299]: 👍🏻

File: ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfigurationFactoryTest.kt [hunk 2]
```diff
@@ -92,18 +92,24 @@ internal class TemplateConfigurationFactoryTest {
...
+            PackageType.WEEKLY -> "$1.99/wk ($7.96/mth)"
+            else -> error("Unknown package type ${rcPackage.packageType}")
+        }
         val offerDetailsWithIntroOffer = when(rcPackage.packageType) {
-            PackageType.ANNUAL -> "PRICE_AND_PER_MONTH after 1 month trial"
-            PackageType.MONTHLY -> "PRICE_AND_PER_MONTH after  trial"
-            PackageType.WEEKLY -> "PRICE_AND_PER_MONTH after  trial"
+            PackageType.ANNUAL -> "$67.99/yr ($5.67/mth) after 1 month trial"
+            PackageType.MONTHLY -> "$7.99/mth after  trial"
+            PackageType.WEEKLY -> "$1.99/wk ($7.96/mth) after  trial"
```
Comment (NachoSoto) [inline ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfigurationFactoryTest.kt:104]: Nit 
```suggestion
            PackageType.MONTHLY -> "$7.99/mth after trial"
            PackageType.WEEKLY -> "$1.99/wk ($7.96/mth) after trial"
```

**Review by NachoSoto — COMMENTED** [review]
Yay!

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt [hunk 3]
```diff
@@ -47,30 +50,30 @@ internal class VariableDataProvider(
...
-    fun localizedPriceAndPerMonth(rcPackage: Package): String {
-        return "PRICE_AND_PER_MONTH"
+    fun localizedPriceAndPerMonth(rcPackage: Package, locale: Locale): String {
+        if (!rcPackage.isSubscription || rcPackage.isMonthly) {
+            return localizedPricePerPeriod(rcPackage, locale)
+        }
+        val unit = Period(1, Period.Unit.MONTH, "P1M").localizedAbbreviatedPeriod(locale)
+        val pricePerPeriod = localizedPricePerPeriod(rcPackage, locale)
+        val pricePerMonth = localizedPricePerMonth(rcPackage, locale)
+        return "$pricePerPeriod ($pricePerMonth/$unit)"
```
Comment (vegaro) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt:76]: not sure if we are focusing on that yet, but does this work fine for for RTL languages?

**Review by vegaro — APPROVED** [review]

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt (hunk 3 above)
Comment (tonidero) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt:76]: Good question... Maybe not, but currently I just copied what we are doing in iOS. We can revisit this though

File: ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfigurationFactoryTest.kt (hunk 2 above)
Comment (tonidero) [reply ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfigurationFactoryTest.kt:104]: This is actually a consequence of using `after {{ sub_offer_duration }} trial` in packages that don't have offers (the monthly and weekly test packages don't have offers, so it leaves an extra space. I could add offers to the test data to get the tests looking better, but this is the real result right now... Will revisit this later and try to improve the tests.

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt (hunk 3 above)
Comment (NachoSoto) [reply ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/VariableDataProvider.kt:76]: Yeah I forgot to ask. iOS handles these strings automatically and makes them work on RTL. We'll test the Arabic paywall before launch.

File: ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfigurationFactoryTest.kt (hunk 2 above)
Comment (NachoSoto) [reply ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/processed/TemplateConfigurationFactoryTest.kt:104]: Ooooh okay that's fine then

**Pull request merged**